extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -163,15 +163,26 @@ private boolean isHandledByProducers(Type type) {
                 case FLOAT:
                 case LONG:
                 case INT:
+                case SHORT:
+                case BYTE:
+                case CHAR:
```

Comment:
Since all primitives are handled here now, you could just `return type.asPrimitiveType().primitive()` instead couldn't you?  Or even just return `true` perhaps?

Reply:
You are right.